refactor: review document partner plugin (#1)
laurodd/docs.konghq.com: app/_hub/datadome/kong-plugin-datadome/_index.md

This plugin will analyse in real time any API requests received. For each request, an api call is sent to DataDome Bot & Fraud Protection Platform to validate if this is legitimate or not.

> The only thing to configure is your DataDome Server Side Key.

Reviewer: Rephrasing to make it a bit shorter.

Also:
- Prefer using US English over UK English
- Capitalize acronyms consistently
- Don't capitalize common names
- Prefer explicit nouns, e.g. "thing" -> "requirement"

```suggestion
This plugin relies on the DataDome Bot & Fraud Protection Platform to validate if any incoming API request is legitimate or coming from a bot.
The only requirement to configure it is your DataDome server-side key.
```

## How it works

> The DataDome plugin will hook into the request/response of an API request and it is executed for every request from a client and before it is being proxied to the upstream on the access phase (see details [here](https://openresty-reference.readthedocs.io/en/latest/Directives/)).

Reviewer: Rephrasing to make it shorter.

```suggestion
The DataDome plugin will hook into every API request from a client, and it will run before they are proxied to the upstream on the access phase (see details [here](https://openresty-reference.readthedocs.io/en/latest/Directives/)).
```

Author: done

## How to install

> Custom plugins can be installed via LuaRocks. A Lua plugin is distributed in `.rock` format, which is
> a self-contained package that can be installed locally or from a remote server.

Reviewer: Rephrasing to clarify what to expect.

```suggestion
Custom plugins can be installed via LuaRocks. A Lua plugin is distributed as a `.rock` file, which is
a self-contained package that can be installed locally or from a remote server.
```

Author: this is kong default text, we can’t change

If you used one of the official {{site.base_gateway}} installation packages, the LuaRocks utility
should already be installed in your system.

> Install the `.rock` in your LuaRocks tree, that is, the directory in which LuaRocks
> installs Lua modules.

Reviewer: Rephrasing to remove unneeded details and focus on guiding the reader instead.

```suggestion
Add the `.rock` package to your LuaRocks tree by following the instructions below.
```

Author: same here, kong default

1. Install the DataDome plugin:

    ```sh
    luarocks install kong-plugin-datadome
    ```

> 2. Update your loaded plugins list in {{site.base_gateway}}.

Reviewer: ```suggestion
2. Update the list of loaded plugins in {{site.base_gateway}}.
```

Author: kong default

>     In your `kong.conf`, append `datadome` to the `plugins` field. Make sure the field is not commented out.

Reviewer: ```suggestion
    In your `kong.conf` file, append `datadome` to the `plugins` field. Make sure the field is not commented out.
```

Author: kong default

    ```yaml

>     plugins = bundled,datadome              # Comma-separated list of plugins this node
>                                             # should load. By default, only plugins
>                                             # bundled in official distributions are
>                                             # loaded via the `bundled` keyword.

Reviewer: The comment would be easier to read if it was not so compact.

```suggestion
    # Comma-separated list of plugins this node should load.
    # By default, only plugins bundled in official distributions are loaded via the `bundled` keyword.
    plugins = bundled,datadome
```

Author: kong default

    ```

3. Restart {{site.base_gateway}}:

    ```sh
    kong restart
    ```

## Using the plugin

> First of all, make sure you have your DataDome Server Side Key. It is available inside your [Dashboard > Integrations](https://app.datadome.co/dashboard/management/integrations).

Reviewer: ```suggestion
First of all, make sure you have your DataDome server-side key. It is available inside your dashboard in the [Integrations](https://app.datadome.co/dashboard/management/integrations) section.
```

Author: done

> #### Kong Gateway (OSS)
> If you already have an API configured, execute the command below replacing the following `<values>`:

Reviewer: - Use correct heading level
- Prefer active form instead of passive
- Use explicit terms instead of `<values>` as a pattern

```suggestion
### Kong Gateway (OSS)
If you already configured an API, execute the command below after replacing `<YOUR_API>` with the name of your API and `<server_side_key>` with your DataDome server-side key.
```

Reviewer: On https://docs.datadome.co/docs/kong-plugin, it's called Kong Open Source.
Which one is the official name?

Author: they use in both ways (https://docs.konghq.com/gateway/latest/#on-premises), but I would update the public doc like here Kong Gateway (OSS)  or Kong Gateway OSS (Open Source) if we want to be really explicit

```shell
curl -i -X POST http://localhost:8001/services/<YOUR_API>/plugins \
     -F "name=datadome" \
     -F "config.datadome_server_side_key=<server_side_key>"

>  ```

Reviewer: Extra space to remove in front of the triple backticks.

Author: done

> #### Kong Konnect

Reviewer: ```suggestion
### Kong Konnect
```

Author: done

> - Depending where you want to enable DataDome, select *Plugins*
> - Click *+ New Plugin*
> - On Custom Plugins, select *DataDome*
> - Fill *Datadome Server Side Key*
> - *Save*

Reviewer: ```suggestion
- Depending on where you want to enable DataDome, select `Plugins`
- Click on `+ New Plugin`
- On `Custom Plugins`, select `DataDome`
- Fill the *DataDome Server-side Key* field
- Hit `Save`
```

Reviewer: Not sure where the `Datadome Server Side Key` label comes from, but it should be updated too.

Author: It comes from kong konnect: it seems that the platform gets the parameters (defined on the schema.lua file) like datadome-server-side-key, excludes the '-' dashes and put all the first letters capitalised, resulting `Datadome Server Side Key`.
Thats the reason.

The official doc is here:
https://dash.readme.com/project/datadome/v6.0/docs/kong-plugin

Author: <img width="389" alt="image" src="https://github.com/laurodd/docs.konghq.com/assets/111757181/7e4d0a13-f4fb-4f18-9aa3-4dd958a14c5d">
